app/forms/work_deposit_pathway.rb

```diff
@@ -33,6 +33,10 @@ def allows_curation_request?
     data_and_code? && !@resource.draft_curation_requested
   end
 
+  def allows_mint_doi_request?
+    data_and_code? && @resource.work.doi.blank? && !DoiMintingStatus.new(@resource.work).present?
```

Comment:
I'd suggest two refactorings here.

The first one is trivial and also much less important, but I think that the code would read a little better if we implemented `DoiMintingStatus#blank?` and used that here instead of `!` and `#present?`.

The second and more important thing is that I'd try to get rid of the method chaining here. `WorkDepositPathway` needs to know too much about the resource object that it's holding. It needs to know that the resource has a work and that the work in turn can have a DOI. It would be better if each type of thing that might act as a resource here (maybe it's only `WorkVersion`? I can't remember..) could handle telling the work deposit pathway whether or not it has a DOI directly - i.e. `@resource.doi_blank?`. That reads better too, but it also relieves `WorkDepositPathway` of needing to know about the structure/hierarchy of that other information.

Reply:
First one I agree though as a note, `#present?` here is not a normal present. It's actually a custom method of `DoiMintingStatus` (threw me off for a long time trying to troubleshoot because `#present?` and `#blank?` were both returning false). But thinking on it, I can also just do a custom `#blank?` on DoiMintingStatus so we can still follow typical conventions.

Second sounds good & I'll get working on that.